packages/react-instantsearch-core/src/connectors/connectVoiceSearch.ts

```diff
@@ -0,0 +1,72 @@
+import createConnector from '../core/createConnector';
+import voiceSearchHelper from '../lib/voiceSearchHelper';
+import { refineValue, getCurrentRefinementValue } from '../core/indexUtils';
+
+function getId() {
+  return 'query';
+}
+
+function refine(props, searchState, nextRefinement, context) {
+  const id = getId();
+  const nextValue = { [id]: nextRefinement };
+  const resetPage = true;
+  return refineValue(searchState, nextValue, context, resetPage);
+}
+
+function getCurrentRefinement(props, searchState, context) {
+  const id = getId();
+  const refinementsCallback = currentRefinement => currentRefinement || '';
+  return getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    id,
+    '',
+    refinementsCallback
+  );
+}
+
+export default Composed => {
+  const helper = voiceSearchHelper();
```

Comment:
I wouldn't call this `helper`, it will be confusing with the algoliasearch helper

Reply:
good point. I've renamed it to `voiceSearch`. Any other suggestion is welcomed.